homework/index.js

```diff
@@ -33,10 +33,81 @@
   function main(url) {
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
+
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        const header = createAndAppend('header', root, { id: 'header' });
+        createAndAppend('p', header, { text: 'HYF Repositories' });
+        const select = createAndAppend('select', header, { id: 'select' });
+
+        // CREATE EVENT LISTENER
+        select.addEventListener(
+          'change',
+          () => {
+            const selected = select.selectedIndex;
+            const repositoryValue = data[selected].name;
+            const repoLink = data[selected].html_url;
+            if (repositoryValue === null) {
+              document.getElementById('property_0').innerText = '-';
+            } else {
+              document.getElementById('property_0').innerText = repositoryValue;
+              document.getElementById('property_0').setAttribute('href', repoLink);
+              document.getElementById('property_0').setAttribute('target', '_blank');
+            }
+            const descriptionValue = data[selected].description;
+            if (descriptionValue === null) {
+              document.getElementById('property_1').innerText = '-';
+            } else {
+              document.getElementById('property_1').innerText = descriptionValue;
+            }
+            const forksValue = data[selected].forks;
+            if (forksValue === null) {
+              document.getElementById('property_2').innerText = '-';
+            } else {
+              document.getElementById('property_2').innerText = forksValue;
+            }
+            const updatedValue = new Date(data[selected].updated_at);
+            if (updatedValue === null) {
+              document.getElementById('property_3').innerText = '-';
+            } else {
+              document.getElementById('property_3').innerText = updatedValue;
+            }
+          },
+          false,
+        );
+
+        // SORT SELECTION MENU ALPHABETICALLY
+        const dataSorted = data.sort((a, b) => a.name.localeCompare(b.name));
```

Comment:
Watch out here: `sort` is a *mutating* function - it means that after this call, the contents of `data` itself are sorted. What it returns (and what you assign to `dataSorted` is literally the same array as `data`).

Reply:
Got it. So, there is no need to create a new variable because my intention is already to mutate the variable.